### PR TITLE
fix: fixed validation errors duplicates

### PR DESCRIPTION
**Pull request by chesterkmr**
### Description
- Added uniqBy to validator output to avoid error duplicates.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced error message filtering to remove duplicates in validation processes.

___

### **PR Type**
Bug fix

___

### **Description**
- Enhanced the JSON Schema Rule Engine to filter out duplicate error messages using lodash's `uniqBy` function.
- This change ensures that each error message is unique, improving the clarity and usefulness of validation error outputs.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>json-schema.rule-engine.ts</strong><dd><code>Implement Unique Error Message Filtering in JSON Schema Rule Engine</code></dd></summary>
<hr>

apps/kyb-app/src/components/organisms/DynamicUI/rule-engines/json-schema.rule-engine.ts
<li>Imported <code>uniqBy</code> from lodash to ensure unique error messages.<br> <li> Modified the error filtering logic to use <code>uniqBy</code> based on the <br>'message' property.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2368/files#diff-27df80121d4f7a5ed418ecc9acd79e0b15d149a1d4ca783fdedf66cd27f6918e">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: apps/kyb-app/src/components/organisms/DynamicUI/rule-engines/json-schema.rule-engine.ts [hunk 1]
```diff
@@ -74,7 +75,10 @@ export class JsonSchemaRuleEngine implements RuleEngine {
       return fieldErrors;
     });
 
-    return result?.flat()?.filter(result => Boolean(result.message));
+    return uniqBy(
+      result?.flat()?.filter(result => Boolean(result.message)),
+      'message',
+    );
```
Comment (Omri-Levy) [inline apps/kyb-app/src/components/organisms/DynamicUI/rule-engines/json-schema.rule-engine.ts:81]: Isn't a message connected to a field? It seems the real issue is that more than one of the same message is being pushed to the same field.

Comment (chesterkmr) [reply apps/kyb-app/src/components/organisms/DynamicUI/rule-engines/json-schema.rule-engine.ts:81]: Duplicate messages are coming from ajv, the way we parsing errors and converting them to field errors didnt handled this case. So uniqBy should be fine.

Comment (Omri-Levy) [reply apps/kyb-app/src/components/organisms/DynamicUI/rule-engines/json-schema.rule-engine.ts:81]: If 3 fields have the same message, would only the first field have the message post this change?

Comment (alonp99) [reply apps/kyb-app/src/components/organisms/DynamicUI/rule-engines/json-schema.rule-engine.ts:81]: Why in the first place we get couple of messages on a single input field? @chesterkmr

**Review by tomer-shvadron — APPROVED** [review]

**Pull request merged**